docs/basic-component.md

```diff
@@ -0,0 +1,247 @@
+# Basic component recipe
+
+Recipe for creating a basic component.
+
+- [Component structure](#component-structure)
+  - [Subcomponents](#subcomponents)
+  - [Hooks and Utils](#hooks-and-utils)
+  - [index.ts](#indexts-1)
+  - [Component.tsx](#componenttsx)
+  - [Styles](#styles)
+  - [Tests](#tests)
+- [Component.tsx template](#componenttsx-template)
+  - [Props](#props)
+  - [The component](#the-component)
+  - [Display name](#display-name)
+
+## Component structure
+
+Given the component `PancakeStack`, the component structure will follow this:
+
+```
+/* Component */
+src/
+  PancakeStack/
+    components/
+      - Pancake/
+        - index.ts
+        - Pancake.tsx
+        - Pancake.scss
+        - Pancake.spec.tsx
+      - Topping/
+        - index.ts
+        - Topping.tsx
+        - Topping.scss
+        - Topping.spec.tsx
+    hooks/
+      - index.ts
+      - useHook.ts
+      - useHook.spec.ts
+    utils/
+      - index.ts
+      - groupedFunctions.ts (eg. getters)
+      - groupedFunctions.spec.ts
+      - functionName.ts
+      - functionName.spec.ts
+    - index.ts
+    - PancakeStack.tsx
+    - PancakeStack.scss
+    - PancakeStack.spec.tsx
+```
+
+### Subcomponents
+
+Subcomponents live in the `components/` directory and should not have subcomponents of their own. If you are finding that you want subcomponents for your subcomponents, then one (or both) should be a component instead.
+
+Aside from the above, subcomponents should follow the same structure as a normal component.
+
+Subcomponents living within their own directories have the benefits of keeping files small and clean, and allows for mocking for tests.
+
+### Hooks and Utils
+
+Hooks and utils go into their own respective folders, with their filename reflective of either the function name or a collective name based on the functionality of functions within the file.
+
+### index.ts
+
+This file is used only as an entrypoint.
+Anything relating to the component (eg. hooks, subcomponents) that are to be exported should be included here.
+
+```ts
+// index.ts
+export * from './PancakeStack';
+export * from './Pancake';
+export * from './hooks';
+```
+
+### Component.tsx
+
+Component files should be named to match the component name (eg. the Pancake component will be named `Pancake.tsx`).
+
+See [template](#componenttsx-template) for component composition.
+
+### Styles
+
+Style files should named to match the component name (eg. the Pancake component will be named `Pancake.scss`) and live in the same directory.
+
+Keep these clean by separating styles for subcomponents in their own respective files.
+
+### Tests
+
+Test files should named to match the component/function name (eg. `Pancake.spec.tsx` and `functionName.ts`) and live in the same directory.
+
+Ideally we would want to avoid snapshot tests as they do not have meaningful assertions/expectations and are very fragile (they can break with insignificant changes). We should instead write unit and functional tests.
```

Comment:
Just wondering if we can have guidelines on what we should be testing here. Eg. interaction testing, accessibility, style changes associated with functionality

Also on how we write tests to be standardised Eg. Given, when, then

Reply:
That will be addressed separately. We have not had a thorough discussion on testing yet.